spyder/utils/syntaxhighlighters.py

```diff
@@ -1074,33 +1075,81 @@ def __init__(self, parent, font=None, color_scheme=None):
         # Load Pygments' Lexer
         if self._lang_name is not None:
             self._lexer = get_lexer_by_name(self._lang_name)
+
+            
+
+        # Connect document updates to a re-lexing of the document.
+        # parent.contentsChange.connect(self._make_charlist)
         BaseSH.__init__(self, parent, font, color_scheme)
+        self._worker_manager = WorkerManager()
+        self._charlist = []
+        self._allow_highlight = True
+
+    def make_charlist(self):
+        """"""
+        def test_output(worker, output, error):

```

Comment:
Why is this function called `test_output`? It would seem to imply that it's only used for tests.